hs/src/LedgerState.hs

```diff
@@ -441,9 +441,19 @@ retirePools ls@(LedgerState _ ds _) epoch =
                & retiring .~ active)
   where (active, retiring') = Map.partition (epoch /=) (ds ^. retiring)
 
+-- |Calculate the change to the deposit pool for a given transaction.
+depositPoolChange :: LedgerState -> Tx -> Coin
+depositPoolChange ls tx = (currentPool + txDeposits) - txRefunds
+  where
+    currentPool = ls ^. utxoState . deposits
+    txDeposits = depositAmount (ls ^. pcs) (ls ^. delegationState . stPools) tx
+    txRefunds = keyRefunds (ls ^. pcs) (ls ^. delegationState . stKeys) tx
+
 -- |Apply a transaction body as a state transition function on the ledger state.
 applyTxBody :: LedgerState -> Tx -> LedgerState
 applyTxBody ls tx = ls & utxoState . utxo .~ newUTxOs
+                       & utxoState . deposits .~ depositPoolChange ls tx
+                       & utxoState . fees .~ (tx ^. txfee) + (ls ^. utxoState . fees)
```

Comment:
make it an update using `%~`

Reply:
ok, I'll try that. I tried using `+~` but I got a parsing error about infix priorities.